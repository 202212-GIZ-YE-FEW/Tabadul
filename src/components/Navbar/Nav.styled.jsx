import styled from "@emotion/styled";
import Link from "next/link";

export const Header = styled.header`
    position: sticky;
    z-index: 10;\n
    width: 100%;
`;

export const Nav = styled.nav`
    display: flex;
    padding: 6px;
    justify-content: space-between;
    align-items: center;
    background-color: #f1f1f1;

    @media only screen and (max-width: 767px) {
     {
          width: 100%;
        }

    
        @media only screen and (min-width: 768px) {
         
              width: 768px;
             
            }
`;

export const Navmenubar = styled.div`
    display: flex;
    flex-direction: column;
    row-gap: 6px;

    @media screen and (min-width: 768px) {
        display: none;
        gap: 10px;
    }
`;

export const Navmenubardiv = styled.div`
    width: 40px;
    height: 8px;
    background-color: #33956d;
    border-radius: 2px;
    cursor: pointer;
`;

export const Navmenulist = styled.div`
    display: flex;
    flex-direction: column;
    position: fixed;
    top: 100px;
    left: 0px;
    width: 100%;
    row-gap: 24px;
    padding: 24px 16px;
    min-height: calc(100vh - 60px);
    text-align: center;
    transition: all 0.2s;
    background-color: #f1f1f1;

    @media screen and (min-width: 768px) {
        position: unset;
        min-height: fit-content;
        width: auto;
        flex-direction: row;
        align-items: center;
        column-gap: 24px;
    }

    @media screen and (max-width: 768px) {
        gap: 10px;
        right: 0;
    }
`;

export const Navlink = styled.a`
    font-size: 18px;
    position: relative;
    text-decoration: none;
    color: #3c4347;
    transition: all 0.2s;
  

    &:hover {
        font-weight: bold;
    }

    @media screen and (min-width: 768px) {
        &:before {
            content: "";
            position: absolute;
            width: 0px;
            bottom: -16px;
            height: 6px;
            transition: all 0.2s;
            background-color: #33956d;
        }
        
        &:hover:before {
            width: 70px;\n    }
    }
    
`;

export const Homeiconlink = styled(Link)`
    text-decoration: none;

    @media screen and (max-width: 768px) {
        gap: 10px;
        display: none;
    }
`;

export const Homeicon = styled.div`
    height: 60px;
    width: 60px;
    padding-left: 30px;
`;

export const IconVector = styled.img`
    &:before {
        padding-left: 26px;
    }
`;

export const Dropbtn = styled.button`
    background-color: transparent;
    color: white;
    font-size: 16px;
    border: none;
    cursor: pointer;
    margin-right: 15px;
    display: flex;
    flex-direction: row;
    align-items: center;
`;

export const Dropdown = styled.div`
    position: relative;
    display: inline-block;
`;

export const Dropdowncontent = styled.div`
    position: absolute;
    right: 0px;
    background-color: #f9f9f9;
    min-width: 160px;
    font-size: 15px;
    box-shadow: 0px 8px 16px 0px rgba(0, 0, 0, 0.2);
    z-index: 1;
`;

export const Lanlink = styled(Link)`
    color: black;
    padding: 12px 16px;
    text-decoration: none;
    display: block;
    font-size: 18 px;
    &:hover {
        background-color: #33956d;
    }
`;

export const Mynav = styled.div`
    display: flex;
    flex-direction: row;
    align-items: center;
`;

export const NavButtons = styled.div`
    display: flex;
    flex-direction: row;

    @media screen and (max-width: 768px) {
        gap: 10px;
    }
`;

export const Imageicon = styled.img`
    height: 50px;
    width: 50px;
`;

export const Arrow = styled.img`
    margin-left: 10px;
    height: 12px;
`;

export const Lan = styled.img`
    border-radius: 20px;
    height: 30px;
    margin-left: 25px;
`;
