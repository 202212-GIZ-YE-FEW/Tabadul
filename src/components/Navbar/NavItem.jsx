import React from "react";

import { Navlink } from "./Nav.styled";

export const NavItem = ({ href, active, text }) => {
    return (
        <Navlink href={href} active={` ${active ? "active" : ""}`}>
            {/* {icon && <span className={`icon-${icon}`} />} */}
            <span dangerouslySetInnerHTML={{ __html: text }} />
        </Navlink>
    );
};
