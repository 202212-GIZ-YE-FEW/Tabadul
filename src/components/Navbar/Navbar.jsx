import { useEffect, useState } from "react";

import {
    Arrow,
    Dropbtn,
    Dropdown,
    Dropdowncontent,
    Header,
    Homeicon,
    Homeiconlink,
    IconVector,
    Imageicon,
    Lan,
    Lanlink,
    Mynav,
    Nav,
    NavButtons,
    Navmenubar,
    Navmenubardiv,
    Navmenulist,
} from "./Nav.styled";
import { NavItem } from "./NavItem";

const MENU_LIST = [
    {
        id: 1,
        text: "Home",

        href: "/",
    },

    { id: 2, text: "About us", href: "Aboutus" },

    { id: 3, text: "Products", href: "Products" },

    { id: 4, text: "Blogs", href: "Blogs" },

    { id: 5, text: "profile", href: "Profile" },
];

export default function Navbar() {
    const [navActive, setNavActive] = useState(false);
    const [activeIdx, setActiveIdx] = useState(false);
    const [navdroplan, setNavdroplan] = useState(false);
    const [navdrop, setNavdrop] = useState(false);
    const [width, setWidth] = useState(0);
    useEffect(() => {
        setWidth(window.innerWidth);
    }, []);
    return (
        <Header>
            <Nav>
                <Homeiconlink href='/'>
                    <Homeicon onClick={() => setActiveIdx(false)}>
                        <IconVector src='/images/Logo.svg' />
                    </Homeicon>
                </Homeiconlink>

                <Navmenubar
                    onClick={() => setNavActive((prevState) => !prevState)}
                    onBlur={() => setNavdrop((prevState) => !prevState)}
                >
                    <Navmenubardiv></Navmenubardiv>
                    <Navmenubardiv></Navmenubardiv>
                    <Navmenubardiv></Navmenubardiv>
                </Navmenubar>
                <Mynav>
                    {(navActive || width > 1000) && (
                        <Navmenulist>
                            {MENU_LIST.map((menu, idx) => {
                                return (
                                    <div
                                        //ToDo: Style this to center it
                                        onClick={() => {
                                            setActiveIdx(idx);
                                            setNavActive(
                                                (prevState) => !prevState
                                            );
                                        }}
                                        key={menu.text}
                                    >
                                        <NavItem
                                            active={activeIdx === idx}
                                            {...menu}
                                        />
                                    </div>
                                );
                            })}
                        </Navmenulist>
                    )}
                    <NavButtons>
                        <Dropdown>
                            <Dropbtn
                                onClick={() => setNavdrop(!navdrop)}
                                onBlur={() =>
                                    setTimeout(() => setNavdrop(false), 100)
                                }
                            >
                                <Imageicon src='/images/users.svg' />
                            </Dropbtn>

                            {navdrop && (
                                <Dropdowncontent>
                                    <Lanlink href='/Signin'>Log in</Lanlink>
                                    <Lanlink href='/Signup'>Sign up</Lanlink>
                                    <Lanlink href='#'>Log out</Lanlink>
                                </Dropdowncontent>
                            )}
                        </Dropdown>
                        <Dropdown>
                            <Dropbtn
                                onClick={() => setNavdroplan(!navdroplan)}
                                onBlur={() => setNavdroplan(!navdroplan)}
                            >
                                <Imageicon src='/images/flag.svg' />
                                <Arrow src='/images/arrow.svg' />
                            </Dropbtn>
                            {navdroplan && (
                                <Dropdowncontent>
                                    <Lanlink href='#'>
                                        English
                                        <Lan src='/images/flag.svg' />
                                    </Lanlink>
                                    <Lanlink href='#'>
                                        Arabic
                                        <Lan src='/images/Saudi Arabia.svg' />
                                    </Lanlink>
                                    <Lanlink href='#'>
                                        Turkish
                                        <Lan src='/images/Turkey.svg' />
                                    </Lanlink>
                                </Dropdowncontent>
                            )}
                        </Dropdown>
                    </NavButtons>
                </Mynav>
            </Nav>
        </Header>
    );
}
