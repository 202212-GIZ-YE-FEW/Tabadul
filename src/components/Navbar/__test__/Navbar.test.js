import renderer from "react-test-renderer";

import Navbar from "../Navbar";

jest.mock("next/router", () => ({
    useRouter() {
        return {
            pathname: "",
            // ... whatever else you you call on `router`
        };
    },
}));

it("renders correctly", () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
});
