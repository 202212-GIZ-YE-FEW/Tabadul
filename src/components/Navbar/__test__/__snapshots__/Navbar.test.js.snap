// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="css-ga8lz6"
>
  <nav
    className="css-97a4l7"
  >
    <a
      className="css-1qj8w5r"
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="css-1q0bv5v"
        onClick={[Function]}
      >
        <img
          className="css-x41lxq"
          src="./images/Logo.svg"
        />
      </div>
    </a>
    <div
      className="css-pkg6m"
      onBlur={[Function]}
      onClick={[Function]}
    >
      <div
        className="css-w9k2l0"
      />
      <div
        className="css-w9k2l0"
      />
      <div
        className="css-w9k2l0"
      />
    </div>
    <div
      className="css-1t7ejvz"
    >
      <div
        className="css-11f3fck"
      >
        <div
          className="css-1c5viz"
        >
          <button
            className="css-17usi8j"
            onBlur={[Function]}
            onClick={[Function]}
          >
            <img
              className="css-gdzbep"
              src="./images/users.svg"
            />
          </button>
        </div>
        <div
          className="css-1c5viz"
        >
          <button
            className="css-17usi8j"
            onBlur={[Function]}
            onClick={[Function]}
          >
            <img
              className="css-gdzbep"
              src="./images/flag.svg"
            />
            <img
              className="css-jcgd1c"
              src="./images/arrow.svg"
            />
          </button>
        </div>
      </div>
    </div>
  </nav>
</header>
`;
